test/functional/tool_signet_miner.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test signet miner tool"""
+
+import os.path
+import subprocess
+import sys
+import time
+
+from test_framework.key import ECKey
+from test_framework.script_util import keys_to_multisig_script
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+from test_framework.wallet_util import bytes_to_wif
+
+
+CHALLENGE_PRIVATE_KEY = (42).to_bytes(32, 'big')
+
+
+class SignetMinerTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.chain = "signet"
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+
+        # generate and specify signet challenge (simple 1-of-1 multisig script)
+        privkey = ECKey()
+        privkey.set(CHALLENGE_PRIVATE_KEY, True)
+        pubkey = privkey.get_pubkey().get_bytes()
+        challenge = keys_to_multisig_script([pubkey])
+        self.extra_args = [[f'-signetchallenge={challenge.hex()}']]
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_cli()
+        self.skip_if_no_wallet()
+        self.skip_if_no_bdb()
```

Comment:
Is there a specific reason to require bdb here? I don't think it even specifically makes a bdb wallet?

Reply:
The problem was the signing in the descriptor wallet rather than the import of the private key. Now it should work with both wallet types: https://github.com/bitcoin/bitcoin/pull/24559#issuecomment-1098546332